addons/barcodes_gs1_nomenclature/models/barcode_nomenclature.py

```diff
@@ -153,7 +153,10 @@ def _preprocess_gs1_search_args(self, args, barcode_types, field='barcode'):
                     data_type = data['rule'].type
                     value = data['value']
                     if data_type in barcode_types:
-                        match = re.match('0*([0-9]+)$', str(value))
+                        if data_type == 'lot':
+                            match = re.match('0*([\w-]+)$', str(value))
+                        else:
+                            match = re.match('0*([0-9]+)$', str(value))
```

Comment:
I think the regex is useless in case of the lot because in this case, the whole value should be the actual lot name/barcode, there is no zero padding (and so, we doesn't even have to replace the search operator)
```suggestion
                        if data_type == 'lot':
                            args[i] = (field_name, operator, value)
                            break
                        match = re.match('0*([0-9]+)$', str(value))
```